### ISSUE TITLE
[MSVC][x86] Failed to build with error C2488: 'naked' can only be applied to non-member function definitions

### ISSUE THREAD

**Issue by NEIL-smtg** [dotnet/runtime#104460]
### Description

Note:
To reproduce this, you'll need wait for VS 2022 17.12 Preview or later version to ship which will contain this change.

Recently, we have been building this project using the latest build of MSVC, and we have encountered the following error:

```c++
runtime\src\coreclr\debug\ee\debugger.cpp(13904) :
error C2488: 'Debugger::GenericHijackFunc': 'naked' can only be applied to non-member function definitions
```

Complete build log: [Build.log](https://github.com/user-attachments/files/16105948/Build.1.log)

### Reproduction Steps

1. Open X86 Native Tools Command Prompt for VS 2022
2. git clone https://github.com/dotnet/runtime.git
3. cd runtime
4. set _CL_=/wd5281
5. .\build.cmd -subset clr+libs -c release -arch x86 2>&1

### Expected behavior

It compiles.

### Actual behavior

```c++
runtime\src\coreclr\debug\ee\debugger.cpp(13904) :
error C2488: 'Debugger::GenericHijackFunc': 'naked' can only be applied to non-member function definitions
```

### Other information
This is the narrowed down repro, test.cpp:

```c++
struct Foo {
    __declspec(naked) void func1() {} // gets C2488
    void func2();
    __declspec(naked) static void func3() {} // gets C2488
};

#if defined(TARGET_X86)
__declspec(naked)
#endif
void Foo::func2() {} // should also get C2488
```

repro command:
```command prompt
cl /c test.cpp /DTARGET_X86
```

For current version, you will only ger error C2488 on line 2 and 4. For the upcoming version of VS, the last line will be also getting the error C2488.

**Comment by huoyaoyuan** [comment]
This was touched in #102834, but it was already `naked` then.

Does it make sense to port the naked function to asm now?

**Comment by jkotas** [comment]
> Does it make sense to port the naked function to asm now?

Yes. Inline assembles has always been a factory for problems. We try to avoid inline asm for any new code.

**Comment by huoyaoyuan** [comment]
Well I want to understand more about this specific function:

https://github.com/dotnet/runtime/blob/a7efcd9ca9255dc9faa8b4a2761cdfdb62619610/src/coreclr/debug/ee/debugger.cpp#L13918-L13960

It just pushes and pops `ebp` frame around the call to `GenericHijackFuncHelper`. What's the requirement around this?

When written as ordinal function, it compiles to the same code of win-x64:

```asm
call GenericHijackFuncHelper
jmp ExceptionNotForRuntime
``` 

On win-arm64, the return-address saving is around the two function calls:

```asm
str lr,[sp,#-0x10]!
bl GenericHijackFuncHelper
bl ExceptionNotForRuntime
ldr lr,[sp],#0x10
ret
```

**Comment by jkotas** [comment]
https://github.com/dotnet/runtime/blob/a7efcd9ca9255dc9faa8b4a2761cdfdb62619610/src/coreclr/debug/di/rsthread.cpp#L3879-L3881 comment talks about this
